src/store/auth.ts

```diff
@@ -0,0 +1,48 @@
+import { getMe } from 'api/user';
+import { User } from 'api/user/entity';
+import { refreshAccessToken } from 'api/user/userApiClient';
+import { atom, useAtom } from 'jotai';
+import { useEffect } from 'react';
+
+const authAtom = atom<User | null>(null);
+
+const initAuthAtom = atom(null, async (get, set) => {
+  const token = {
+    access: sessionStorage.getItem('accessToken'),
+    refresh: localStorage.getItem('refreshToken'),
+  };
+
+  if (token.access) {
+    const authResponse = await getMe();
+    if (authResponse.data) return set(authAtom, authResponse.data);
+  }
+
+  if (token.refresh) {
+    const refreshResponse = await refreshAccessToken();
+    if (refreshResponse.accessToken) {
+      sessionStorage.setItem('accessToken', refreshResponse.accessToken);
+
+      const authResponse = await getMe();
+      if (authResponse.data) return set(authAtom, authResponse.data);
+    }
+  }
+
+  return set(authAtom, null);
+});
+
+export const useAuthAtom = () => {
+  const [auth] = useAtom(authAtom);
```

Comment:
사소하지만, useAtomValue 를 쓰면 deconstructure 를 하지 않아도 됩니다.

```suggestion
  const auth = useAtomValue(authAtom);
```

Reply:
오..불편한 부분이었는데 감사합니다 ㅋㅋ `useAtomValuse, useSetAtom` 두 가지 메소드 잘 애용할게요!